content-tools/update-cdids/main.go

```diff
@@ -0,0 +1,486 @@
+			continue
+		}
+
+		if len(existingCollectionID) > 0 {
+			pair.Print(ctx, fmt.Sprintf("stopping. Error occurred. CDID: %s already exists in another collection: %s", pair.oldCdID, existingCollectionID))
+			continue
+		}
+
+		cdIDData, err := fetchDataForCDID(ctx, httpClient, environment, oldCdIDLocation)
+		if err != nil {
+			pair.Print(ctx, "stopping. Error occurred while fetching cdID data.")
+			continue
+		}
+
+		err = addCDIDToCollection(ctx, httpClient, environment, collectionID, oldCdIDLocation, cdIDData)
+		if err != nil {
+			pair.Print(ctx, "stopping. Error occurred while adding CDID to collection")
+			continue
+		}
+
+		err = approveCDID(ctx, httpClient, environment, collectionID, oldCdIDLocation)
+		if err != nil {
+			pair.Print(ctx, "stopping. Error occurred while approving cdid in the collection")
+			continue
+		}
+
+		pair.Print(ctx, "Completed Processing ")
+	}
+
+	log.Event(ctx, "successfully updated all documents.", log.INFO)
+}
+
+func checkIfCdIDExistsInAnotherCollection(ctx context.Context, client *http.Client, environment string, cdIDLocation string) (string, error) {
+	cdIDURI := strings.Replace(cdIDLocation, environment, "", -1)
+	collectionCheckURL := fmt.Sprintf("%s/checkcollectionsforuri?uri=%s/data.json", environment, cdIDURI)
+
+	req, err := http.NewRequestWithContext(ctx, "GET", collectionCheckURL, nil)
+	if err != nil {
+		errMessage := fmt.Errorf("failed to prepare search cdid in collection request. Error: %v", err)
+		log.Error(errMessage)
+		return "", errMessage
+	}
+	resp, err := client.Do(req)
+	if err != nil {
+		errMessage := fmt.Errorf("failed to search in API for cdid in collection . Error: %v", err)
+		log.Error(errMessage)
+
+		return "", errMessage
+	}
+	defer resp.Body.Close()
+
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		errMessage := fmt.Errorf("failed to read existing collection ID. Error: %v", err)
+		log.Error(errMessage)
+		return "", errMessage
+	}
+
+	return string(body), nil
+}
+
+// 	curl -X POST --header "X-Florence-Token:$ZEBEDEE_TOKEN"
+//	http://localhost:8082/review/{collection-id}?uri={path-to-content}/data.json&recursive=false
+func approveCDID(ctx context.Context, client *http.Client, environment string, collectionID string, cdIDLocation string) error {
+	cdIDURI := strings.Replace(cdIDLocation, environment, "", -1)
+
+	pageReviewURL := fmt.Sprintf("%s/review/%s?uri=%s/data.json&recursive=false", environment, collectionID, cdIDURI)
+	req, err := http.NewRequestWithContext(ctx, "POST", pageReviewURL, nil)
+	if err != nil {
+		errMessage := fmt.Errorf("failed to approve CDID. Error: %v", err)
+		log.Error(errMessage)
+		return errMessage
+	}
+	req.Header.Set("Content-Type", "application/json")
+	req.Header.Set(FlorenceToken, ctx.Value(ZebedeeToken).(string))
+
+	resp, err := client.Do(req)
+	if err != nil {
+		errMessage := fmt.Errorf("failed to approve CDID. Error: %v", err)
+		log.Error(errMessage)
+
+		return errMessage
+	}
+	if resp.StatusCode > 400 {
```

Comment:
I think this check should be if != 200. Anything other than a 200 OK response status means something when wrong. I think its better to be specific about that status you are expecting back.